Mathlib/Combinatorics/SimpleGraph/Sum.lean

```diff
@@ -73,4 +74,81 @@ def Embedding.sumInr : H ↪g G ⊕g H where
   inj' u v := by simp
   map_rel_iff' := by simp
 
+/-- Color `G ⊕g H` with colorings of `G` and `H` -/
+def Coloring.sum (cG : G.Coloring γ) (cH : H.Coloring γ) : (G ⊕g H).Coloring γ := Coloring.mk
+  (Sum.elim cG cH) <| by
+  intro u v
+  cases u <;> cases v <;> simp
```

Comment:
```suggestion
  rintro (u | u) (v | v) <;> simp
```
`rintro` can be used to make lots of proofs in this file far less verbose. For example

```
def Iso.sumAssoc {I : SimpleGraph γ} : (G ⊕g H) ⊕g I ≃g G ⊕g (H ⊕g I) where
  __ := Equiv.sumAssoc α β γ
  map_rel_iff' := by rintro ((u | u) | u) ((v | v) | v) <;> simp
```